swing-lib/src/test/java/org/triplea/swing/SwingActionTest.java

```diff
@@ -44,89 +46,75 @@
   private Consumer<KeyEvent> consumer;
 
   @Test
-  public void testActionOf() {
+  void testActionOf() {
     final Action action = SwingAction.of("Name1234", listener);
     assertEquals("Name1234", action.getValue(Action.NAME));
     action.actionPerformed(event);
     verify(listener).actionPerformed(event);
   }
 
   @Test
-  public void testKeyReleaseListener() {
+  void testKeyReleaseListener() {
     final KeyEvent event = mock(KeyEvent.class);
     final KeyListener action = SwingAction.keyReleaseListener(consumer);
     action.keyReleased(event);
     verify(consumer).accept(event);
   }
 
   @Test
-  public void testInvokeAndWait_ShouldInvokeActionWhenCalledOffEdt()
-      throws Exception {
+  void testInvokeAndWait_ShouldInvokeActionWhenCalledOffEdt() throws Exception {
     SwingAction.invokeAndWait(action);
 
     verify(action).run();
   }
 
   @Test
-  public void testInvokeAndWait_ShouldInvokeActionWhenCalledOnEdt()
-      throws Exception {
-    SwingUtilities.invokeAndWait(() -> {
-      try {
-        SwingAction.invokeAndWait(action);
-      } catch (final InterruptedException e) {
-        throw new RuntimeException(e);
-      }
-    });
+  void testInvokeAndWait_ShouldInvokeActionWhenCalledOnEdt() throws Exception {
+    SwingUtilities.invokeAndWait(() -> assertDoesNotThrow(() -> SwingAction.invokeAndWait(action)));
 
     verify(action).run();
   }
 
   @Test
-  public void testInvokeAndWait_ShouldRethrowActionUncheckedExceptionWhenCalledOffEdt() {
+  void testInvokeAndWait_ShouldRethrowActionUncheckedExceptionWhenCalledOffEdt() {
     assertThrows(IllegalStateException.class, () -> SwingAction.invokeAndWait(RUNNABLE_THROWING_EXCEPTION));
   }
 
   @Test
-  public void testInvokeAndWait_ShouldRethrowActionUncheckedExceptionWhenCalledOnEdt() throws Exception {
+  void testInvokeAndWait_ShouldRethrowActionUncheckedExceptionWhenCalledOnEdt() throws Exception {
     SwingUtilities.invokeAndWait(() -> assertThrows(
         IllegalStateException.class,
         () -> SwingAction.invokeAndWait(RUNNABLE_THROWING_EXCEPTION)));
   }
 
   @Test
-  public void testInvokeAndWaitResult_ShouldReturnActionResultWhenCalledOffEdt() throws Exception {
+  void testInvokeAndWaitResult_ShouldReturnActionResultWhenCalledOffEdt() throws Exception {
     assertEquals(VALUE, SwingAction.invokeAndWaitResult(() -> VALUE));
   }
 
   @Test
-  public void testInvokeAndWaitResult_ShouldReturnActionResultWhenCalledOnEdt() throws Exception {
-    SwingUtilities.invokeAndWait(() -> {
-      try {
-        SwingAction.invokeAndWaitResult(() -> VALUE);
-      } catch (final InterruptedException e) {
-        throw new RuntimeException(e);
-      }
-    });
+  void testInvokeAndWaitResult_ShouldReturnActionResultWhenCalledOnEdt() throws Exception {
+    SwingUtilities.invokeAndWait(() -> assertDoesNotThrow(() -> SwingAction.invokeAndWaitResult(() -> VALUE)));
```

Comment:
I think there may have been a copy/paste error when I first introduced this test.  Based on the test case name, it should primarily be testing that the expected value is returned from the method under test.  I would suggest a small change to make the test conformant:

```diff
diff --git a/swing-lib/src/test/java/org/triplea/swing/SwingActionTest.java b/swing-lib/src/test/java/org/triplea/swing/SwingActionTest.java
index 47df40b6a..0ee8355ba 100644
--- a/swing-lib/src/test/java/org/triplea/swing/SwingActionTest.java
+++ b/swing-lib/src/test/java/org/triplea/swing/SwingActionTest.java
@@ -13,6 +13,7 @@ import java.awt.event.KeyEvent;
 import java.awt.event.KeyListener;
 import java.time.Duration;
 import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
@@ -94,7 +95,12 @@ class SwingActionTest {
 
   @Test
   void testInvokeAndWaitResult_ShouldReturnActionResultWhenCalledOnEdt() throws Exception {
-    SwingUtilities.invokeAndWait(() -> assertDoesNotThrow(() -> SwingAction.invokeAndWaitResult(() -> VALUE)));
+    final AtomicReference<Object> actualValueRef = new AtomicReference<>();
+
+    SwingUtilities.invokeAndWait(
+        () -> assertDoesNotThrow(() -> actualValueRef.set(SwingAction.invokeAndWaitResult(() -> VALUE))));
+
+    assertEquals(VALUE, actualValueRef.get());
   }
 
   @Test
```

Reply:
Seems reasonable.
Would you mind commiting your patch to the PR and merge afterwards?
I already shut down my dev machine for today.